### PR TITLE
[Merged by Bors] - move: Rearrange basic algebraic subobject

### PR DESCRIPTION
**Pull request by YaelDillies**
For `Foo` an algebraic typeclass, the basic theory of `Subfoo` currently is under `FooTheory.Subfoo`. This is in contradiction to other kinds of subobjects, eg order theoretic ones.

This PR moves that basic theory to `Algebra.Foo.Subfoo` instead.

[Zulip](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Location.20of.20algebraic.20subobjects)

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by eric-wieser** [comment]
> * `FieldTheory.Subfield` to `Algebra.Field.Subfield` because there is nothing field-theoretic happening there

This isn't consistent with how `Subring`, `Submonoid`, `Subgroup` are handled; all reside in the corresponding `*Theory` folder

**Comment by YaelDillies** [comment]
Good point. I'm not sure I like the way subobjects are dealt with, then.

But feel free to do whatever you want with this PR. I'm not fussed.

File: Mathlib/Topology/Algebra/Field/Subfield.lean [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+This file constructs the topological closure of a subfield as a subfield.
+-/
+
+namespace Subfield
+variable {α : Type*} [Field α] [TopologicalSpace α] [TopologicalDivisionRing α]
+
+/-- The (topological-space) closure of a subfield of a topological field is
+itself a subfield. -/
+def topologicalClosure (K : Subfield α) : Subfield α where
```
Comment (eric-wieser) [inline Mathlib/Topology/Algebra/Field/Subfield.lean:20]: Note that `Subring.topologicalClosure` is not split to its own file; so this doesn't seem very beneficial to me either.

**Review by eric-wieser — DISMISSED** [review]
My inclination would be to close this. If we're going to split things relating to subobjects, we should try to be consistent across all subobjects in the way we do so. This PR just makes things less consistent.

**Comment by eric-wieser** [comment]
> Good point. I'm not sure I like the way subobjects are dealt with, then.

Perhaps worth a Zulip thread?

**Comment by semorrison** [comment]
If Eric wants to nix the move, that doesn't seem any obstacle to the main point of this PR, which is to split `Topology.Algebra.Field`, no?

**Comment by eric-wieser** [comment]
> which is to split `Topology.Algebra.Field`

I'm not opposed to this, but I don't think we should do it without also splitting `Topology.Algebra.Ring` to not mention `Subring`, etc

**Comment by semorrison** [comment]
> I'm not opposed to this, but I don't think we should do it without also splitting `Topology.Algebra.Ring` to not mention `Subring`, etc


Seems like a recipe for not getting anything done!

**Comment by eric-wieser** [comment]
> Seems like a recipe for not getting anything done!

Yes, I think leaving files unsplit is better than splitting them inconsistently; the latter just creates more work downstream and makes it harder to work out where to put things

**Comment by YaelDillies** [comment]
This PR is now about moving basic algebraic subobject material from the `FooTheory` folders to `Algebra.Foo.SubFoo`.

**Comment by eric-wieser** [comment]
Once you fix the build, please do a Zulip poll. The diff looks faithful to the intent, but the intent might be contrversial.

**Comment by eric-wieser** [comment]
> Once you fix the build, please do a Zulip poll. The diff looks faithful to the intent, but the intent might be contrversial.

Marking as awaiting author; please make this poll / thread and link it from the description.

**Comment by YaelDillies** [comment]
It's already in my Zulip drafts 😄

**Comment by YaelDillies** [comment]
https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Location.20of.20algebraic.20subobjects

File: Mathlib/Deprecated/Subgroup.lean [hunk 2]
```diff
@@ -16,7 +16,7 @@ This file is deprecated, and is no longer imported by anything in mathlib other
 deprecated files, and test files. You should not need to import it.
 
 This file defines unbundled multiplicative and additive subgroups. Instead of using this file,
-please use `Subgroup G` and `AddSubgroup A`, defined in `GroupTheory.Subgroup.Basic`.
+please use `Subgroup G` and `AddSubgroup A`, defined in `Algebra.Group.Subgroup.Basic`.
```
Comment (Ruben-VandeVelde) [inline Mathlib/Deprecated/Subgroup.lean:19]:
```suggestion
please use `Subgroup G` and `AddSubgroup A`, defined in `Mathlib.Algebra.Group.Subgroup.Basic`.
```

File: Mathlib/Deprecated/Subring.lean [hunk 3]
```diff
@@ -16,7 +16,7 @@ This file is deprecated, and is no longer imported by anything in mathlib other
 deprecated files, and test files. You should not need to import it.
 
 This file defines predicates for unbundled subrings. Instead of using this file, please use
-`Subring`, defined in `RingTheory.Subring.Basic`, for subrings of rings.
+`Subring`, defined in `Algebra.Ring.Subring.Basic`, for subrings of rings.
```
Comment (Ruben-VandeVelde) [inline Mathlib/Deprecated/Subring.lean:19]:
```suggestion
`Subring`, defined in `Mathlib.Algebra.Ring.Subring.Basic`, for subrings of rings.
```

File: Counterexamples/CanonicallyOrderedCommSemiringTwoMul.lean [hunk 4]
```diff
@@ -4,7 +4,7 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Damiano Testa
 -/
 import Mathlib.Data.ZMod.Basic
-import Mathlib.RingTheory.Subsemiring.Order
+import Mathlib.Algebra.Ring.Subsemiring.Order
 import Mathlib.Algebra.Order.Monoid.Basic
```
Comment (Ruben-VandeVelde) [inline Counterexamples/CanonicallyOrderedCommSemiringTwoMul.lean:8]:
```suggestion
import Mathlib.Algebra.Order.Monoid.Basic
import Mathlib.Algebra.Ring.Subsemiring.Order
```

**Review by Ruben-VandeVelde — APPROVED** [review]

**Comment by Ruben-VandeVelde** [comment]
maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Comment by Vierkantor** [comment]
bors d+

**Comment by Vierkantor** [comment]
Let's try that again:

bors merge

**Pull request closed without merging**